tests/cpp/tree/test_gpu_hist.cu

```diff
@@ -275,7 +275,8 @@ void TestHistogramIndexImpl() {
   int constexpr kNRows = 1000, kNCols = 10;
 
   // Build 2 matrices and build a histogram maker with that
-  tree::GPUHistMakerSpecialised<GradientPairPrecise> hist_maker, hist_maker_ext;
+  tree::GPUHistMakerSpecialised<GradientPairPrecise> hist_maker{{ObjInfo::kRegression, true}},
+      hist_maker_ext{{ObjInfo::kRegression, true}};
```

Comment:
Should all C++ tests use an objective with identical Hessians?

Reply:
@hcho3 Let me set it to false by default since that's what most of the objectives have.